it/common/src/main/java/org/apache/beam/it/common/AbstractPipelineLauncher.java

```diff
@@ -0,0 +1,281 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.beam.it.common;
+
+import static org.apache.beam.it.common.logging.LogStrings.formatForLogging;
+import static org.apache.beam.it.common.utils.RetryUtil.clientRetryPolicy;
+
+import com.google.api.client.util.ArrayMap;
+import com.google.api.services.dataflow.Dataflow;
+import com.google.api.services.dataflow.model.Environment;
+import com.google.api.services.dataflow.model.Job;
+import com.google.api.services.dataflow.model.MetricUpdate;
+import dev.failsafe.Failsafe;
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.concurrent.TimeUnit;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
```

Comment:
Other files use a Google `Pattern` class, but here the one that comes with Java is used.

Reply:
Fixed in https://github.com/GoogleCloudPlatform/DataflowTemplates/pull/710.